src/Symfony/Component/ExpressionLanguage/Parser.php

```diff
@@ -87,34 +90,43 @@ public function __construct(
      * variable 'container' can be used in the expression
      * but the compiled code will use 'this'.
      *
+     * @param int-mask-of<Parser::IGNORE_*> $flags
+     *
      * @throws SyntaxError
      */
-    public function parse(TokenStream $stream, array $names = []): Node\Node
+    public function parse(TokenStream $stream, array $names = [], int $flags = 0): Node\Node
     {
-        $this->lint = false;
-
-        return $this->doParse($stream, $names);
+        return $this->doParse($stream, $names, $flags);
     }
 
     /**
      * Validates the syntax of an expression.
      *
      * The syntax of the passed expression will be checked, but not parsed.
-     * If you want to skip checking dynamic variable names, pass `null` instead of the array.
+     * If you want to skip checking dynamic variable names, pass `Parser::IGNORE_UNKNOWN_VARIABLES` instead of the array.
+     *
+     * @param int-mask-of<Parser::IGNORE_*> $flags
      *
      * @throws SyntaxError When the passed expression is invalid
      */
-    public function lint(TokenStream $stream, ?array $names = []): void
+    public function lint(TokenStream $stream, ?array $names = [], int $flags = 0): void
     {
-        $this->lint = true;
-        $this->doParse($stream, $names);
+        if (null === $names) {
+            trigger_deprecation('symfony/expression-language', '7.1', 'Passing "null" as the second argument of "%s()" is deprecated, pass "self::IGNORE_UNKNOWN_VARIABLES" instead as a third argument.', __METHOD__);
+
+            $flags |= self::IGNORE_UNKNOWN_VARIABLES;
+            $names = [];
+        }
+
+        $this->doParse($stream, $names, $flags);
     }
 
     /**
      * @throws SyntaxError
```

Comment:
I suggest adding the phpdoc tag for the flags here as well (and on the property). This way, the psalm job running in our CI can benefit from the more precise type when analyzing the body of the method

Reply:
Done